File: tests/fixtures/supabase/supabase-no-user-metadata-authz-fixed/01-app-metadata-only.ts
import type { User } from '@supabase/supabase-js';

type UserRole = 'student' | 'sponsor';

export function getUserRole(user: User | null): UserRole {
  if (!user) return 'student';
  const role = user.app_metadata?.role;
  return role === 'sponsor' ? 'sponsor' : 'student';
}

File: tests/fixtures/supabase/supabase-no-user-metadata-authz-fixed/01_app_metadata_only.py
from lib.client import supabase


def is_admin(user) -> bool:
    # app_metadata is server-writable only — safe to gate authorization on.
    return user.app_metadata.get("role") == "admin"


def sign_up_with_profile_data(email: str, password: str, display_name: str):
    return supabase.auth.sign_up(
        {
            "email": email,
            "password": password,
            "options": {"data": {"display_name": display_name}},
        }
    )
